lib/node/storage.js

```diff
@@ -1,15 +1,98 @@
 /* eslint no-unused-vars: 0 */
+import { readFile, writeFile } from "fs";
+import * as lockfile from "proper-lockfile";
+import * as path from "path";
 
-export const canStoreURLs = false;
+export const canStoreURLs = true;
+const defaultStoragePath = path.join(process.cwd(), "tus-url-storage.json");
 
-export function setItem(key, value) {
+export class FileStorage {
+  constructor(filePath) {
+    this.path = filePath || defaultStoragePath;
+  }
 
-}
+  setItem(key, value, cb) {
+    lockfile.lock(this.path, this._lockfileOptions).then((release) => {
+      cb = this._releaseAndCb(release, cb);
+      this._getData((err, data) => {
+        if (err) {
+          return cb(err);
+        }
 
-export function getItem(key) {
+        data[key] = value;
+        this._writeData(data, (err) => cb(err));
+      });
+    }).catch(cb);
+  }
 
-}
+  getItem(key, cb) {
+    this._getData((err, data) => {
+      if (err) {
+        return cb(err);
+      }
+      cb(null, data[key]);
+    });
+  }
+
+  removeItem(key, cb) {
+    lockfile.lock(this.path, this._lockfileOptions).then((release) => {
+      cb = this._releaseAndCb(release, cb);
+      this._getData((err, data) => {
+        if (err) {
+          return cb(err);
+        }
+
+        delete data[key];
+        this._writeData(data, (err) => cb(err));
+      });
+    }).catch(cb);
+  }
 
-export function removeItem(key) {
+  get _lockfileOptions() {
```

Comment:
Please don't use private getters for static objects. I think that's causing more confusion than it's bringing benefit.